component/common/src/main/java/org/wso2/carbon/extension/identity/emailotp/common/EmailOtpServiceImpl.java

```diff
@@ -95,6 +95,16 @@ public GenerationResponseDTO generateEmailOTP(String userId) throws EmailOtpExce
                     String.format("Error while retrieving user for the Id : %s.", userId), e);
         }
 
+        // Check if the user is locked.
+        if (Utils.isAccountLocked(user)) {
+            throw Utils.handleClientException(Constants.ErrorMessage.CLIENT_ACCOUNT_LOCKED, user.getUserID());
```

Comment:
don't we need to use a generic msg unless it was configured to use detailed error msg?

Reply:
fixed with dc2b353916247cd9e18080d6553dc4dfe86bc22b